### PR TITLE
Fix test failures in JSFS: store data to be written in WriteResult's ctor

### PR DESCRIPTION
**Pull request by ussuri**
@rpaquay 

This is a minimal fix for JSFS. All other FSes can also benefit from this, but I didn't want to muddy this PR with those changes; I'm in the process of implementing them, though, and almost done with that. Will upload tomorrow.

### CODE REVIEW

**Comment by ussuri** [comment]
@rpaquay: all tests are passing

File: lib/axiom/fs/stream/stub_file_system.test.js [hunk 1]
```diff
@@ -328,7 +328,7 @@ describe('StubFileSystem', function () {
             expect(writeResult).toBeDefined();
             expect(writeResult.dataType).toBe(DataType.Value);
             // TODO(rpaquay): Shoud this be `data` instead?
-            expect(writeResult.data).toBe(null);
+            expect(writeResult.data).toBe('This is another text file');
```
Comment (rpaquay) [inline lib/axiom/fs/stream/stub_file_system.test.js]: nit: remove the "todo" comment and use the `data` variable (versus hard-coding the string value)

**Comment by rpaquay** [comment]
lgtm after nit

File: lib/axiom/fs/stream/stub_file_system.test.js (hunk 1 above)
Comment (ussuri) [inline lib/axiom/fs/stream/stub_file_system.test.js]: Oh, right. Too hasty fixing :)

**Pull request merged**